service-workers/service-worker/resources/extendable-event-async-waituntil.js

```diff
@@ -43,7 +43,7 @@ self.addEventListener('message', function(event) {
         async_task_waituntil(event).then(reportResultExpecting('InvalidStateError'));
         break;
       case 'script-extendable-event':
-        new_event_waituntil().then(reportResultExpecting('InvalidStateError'));
+        Promise.resolve().then(new_event_waituntil);
```

Comment:
I thought if it goes all sync from dispatching the event to calling port.postMessage(), the event will not be caught by the listener in html as it'd be posted even before the 'ACK' message. But it turns out the event to the port is caught. (In the follow-up patch, the postMessage is done in then callback. But even if it's called synchronously, the even is still caught.)

I reused sync_waituntil() and changed the name of the event to nontrustedevent as well.